website/versioned_docs/version-0.1.0/Connection/3.mongodb_connection_registry.md

```diff
@@ -0,0 +1,133 @@
+---
+id: mongodb_connection_registry
+title: MongoDB Connection Registry
+sidebar_label: MongoDB Connection Registry
+---
+
+
+`MongoDBConnection` provide the `mongo_client` on every call to the `get` method. And will `close` the connection when exiting the `"with"` statement.
+
+## Example
+##### config.yaml
+
+```yaml
+core_lib:
+  mongodb:
+    _target_: core_lib.connection.mongodb_connection_registry.MongoDBConnectionRegistry
+    config:
+      url:
+        protocol: mongodb+srv
+        host: server.example.com
+        port: 27017
+        username: ${oc.env:MONGO_USERNAME}
+        password: ${oc.env:MONGO_PASSWORD}
+```
+
+##### your_core_lib.py
+
+```python
+from core_lib.connection.mongodb_connection_registry import MongoDBConnectionRegistry
+
+
+class YourCoreLib(CoreLib):
+    
+    def __init__(self, conf: DictConfig):
+        self.config = conf.core_lib.mongodb
+        mongodb = MongoDBConnectionRegistry(self.config)
+        self.user = UserDataAccess(mongodb)
+```
+
+##### user_data_access.py
+
+```python
+class UserDataAccess(DataAccess):
+
+    def __init__(self, db: MongoDBConnectionRegistry):
+        self.mongodb = db    
+
+    def get(self, user_id: int) -> Cursor
+        with self.mongodb.get() as client:
+            return client.db.collection.find_one({'_id': user_id})
+  
+    def create(self, data: dict) -> Cursor:
+        with self.mongodb.get() as client:
+            return client.db.collection.insert_one(data)
+
+    def update(self, user_id: int, data: dict):
+        with self.mongodb.get() as client:
+            return client.db.collection.update_one({'_id': user_id}, {'$set': data})

```

Comment:
the function is returning something? 
Is there a return needed? 
what is the type of the returned variable?

Reply:
I think we can return the results just from get because we need data from db. The return type will be a pymongo cursor object and it can be converted to list and for insert, update and delete we will not return anything. Thank you